package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -0,0 +1,821 @@
+             `debug` toggles the debug status for :class:`WaterBridgeAnalysis`,
+             namely the value of :attr:`WaterBridgeAnalysis.debug`.
+
+        See Also
+        --------
+        :meth:`WaterBridgeAnalysis.generate_table` :
+               processing the data into a different format.
+        """
+        logger.info("WBridge analysis: starting")
+        logger.debug("WBridge analysis: donors    %r", self.donors)
+        logger.debug("WBridge analysis: acceptors %r", self.acceptors)
+        logger.debug("WBridge analysis: water bridge %r", self.water_selection)
+
+        remove_duplicates = kwargs.pop('remove_duplicates', True)  # False: old behaviour
+        if not remove_duplicates:
+            logger.warning("Hidden feature remove_duplicates=False activated: you will probably get duplicate H-bonds.")
+
+        debug = kwargs.pop('debug', None)
+        if debug is not None and debug != self.debug:
+            self.debug = debug
+            logger.debug("Toggling debug to %r", self.debug)
+        if not self.debug:
+            logger.debug("WBridge analysis: For full step-by-step debugging output use debug=True")
+
+        self._timeseries = []
+        self.timesteps = []
+
+        logger.info("checking trajectory...")  # n_frames can take a while!
+        try:
+            frames = np.arange(self.u.trajectory.n_frames)[self.traj_slice]
+        except:
+            logger.error("Problem reading trajectory or trajectory slice incompatible.")
+            logger.exception()
+            raise
+        verbose = _set_verbose(verbose=kwargs.get('verbose', None),
+                               quiet=kwargs.get('quiet', None),
+                               default=True)
+        pm = ProgressMeter(len(frames),
+                           format="WBridge frame {current_step:5d}: {step:5d}/{numsteps} [{percentage:5.1f}%]\r",
+                           verbose=verbose)
+
+        try:
+            self.u.trajectory.time
+            def _get_timestep():
+                return self.u.trajectory.time
+            logger.debug("WBridge analysis is recording time step")
+        except NotImplementedError:
+            # chained reader or xyz(?) cannot do time yet
+            def _get_timestep():
+                return self.u.trajectory.frame
+            logger.warning("WBridge analysis is recording frame number instead of time step")
+
+        logger.info("Starting analysis (frame index start=%d stop=%d, step=%d)",
+                    (self.traj_slice.start or 0),
+                    (self.traj_slice.stop or self.u.trajectory.n_frames), self.traj_slice.step or 1)
+
+        for progress, ts in enumerate(self.u.trajectory[self.traj_slice]):
+            # all bonds for this timestep
+
+            # dict of tuples (atom.index, atom.index) for quick check if
+            # we already have the bond (to avoid duplicates)
+
+            frame = ts.frame
+            timestep = _get_timestep()
```

Comment:
`timestep = frame.ts`

Reply:
@richardjgowers Thank you. I have changed the timestep to ts.time.